### PR TITLE
fix: do not show incorrectly support file migration step

### PR DESCRIPTION
**Pull request by estrada9166**
- Closes  #22012 

### User facing changelog

We were showing the supportFile migration step, based on a default `supportFile` which was a hardcoded string, making the migration flow, show this step even if the custom supportFile set on the config file includes it partially, like for example "repo/src/platform/testing/e2e/cypress/support/index.js"

### Steps to test

1. Create a supportFile in a path that is not the default but includes `cypress/support/index.`
2. Run the migration

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/data-context/src/sources/migration/utils.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import _ from 'lodash'
+
+export const isDefaultSupportFile = (supportFile: string) => {
+  if (_.isNil(supportFile) || !_.isBoolean(supportFile) && supportFile.match(/^cypress\/support($|\/index($|\.(ts|js|coffee)$))/)) {
```
Comment (estrada9166) [inline packages/data-context/src/sources/migration/utils.ts:4]: Added the `coffee` ext here

**Review by tgriesser — APPROVED** [review]

File: packages/launchpad/cypress/e2e/migration.cy.ts [hunk 2]
```diff
@@ -503,6 +503,40 @@ describe('Full migration flow for each project', { retries: { openMode: 0, runMo
     checkOutcome()
   })
 
+  it('completes journey for migration-e2e-custom-supportFile', () => {
```
[the remark this answers is not in the document]
Comment (flotwig) [reply packages/launchpad/cypress/e2e/migration.cy.ts:506]: This is testing a custom support file which doesn't exist. #22012 is a custom support file which does exist. Unless I'm missing something, you need a test case for "a custom support file which does exist"

[the remark this answers is not in the document]
Comment (flotwig) [reply packages/launchpad/cypress/e2e/migration.cy.ts:506]: In his case, I'd expect these to all fail:

```
    // supportFile is false - cannot migrate
    cy.get(renameSupportStep).should('not.exist')
    cy.get(setupComponentStep).should('not.exist')
    cy.get(configFileStep).should('exist')
```

File: packages/data-context/src/sources/migration/utils.ts (hunk 1 above)
Comment (ZachJW34) [inline packages/data-context/src/sources/migration/utils.ts:4]: I know this code was lifted, but what happens if the user has `./cypress/support/index.js` (prefixed with "./"). This is the default file but the regex wouldn't match

Comment (estrada9166) [reply packages/data-context/src/sources/migration/utils.ts:4]: Maybe we should update the regex for something like `(^|.+/)cypress\/support($|\/index($|\.(ts|js|coffee)$))` what do you think?

Comment (estrada9166) [reply packages/data-context/src/sources/migration/utils.ts]: [2e5b5d6](https://github.com/cypress-io/cypress/pull/22015/commits/2e5b5d6c9bda0efd3245056f9746e920e00a5b93)

**Review by flotwig — APPROVED** [review]

**Pull request merged**